data/src/main/java/com/antosik/benchproject/data/movie/details/entity/MovieDetailsEntity.kt

```diff
@@ -0,0 +1,30 @@
+package com.antosik.benchproject.data.movie.details.entity
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import com.antosik.benchproject.domain.movie.details.entity.MovieDetailsModel
+
+@Entity
+internal data class MovieDetailsEntity(
+    @PrimaryKey(autoGenerate = false)
+    val movieId: Int,
+    val imageBackgroundPath: String,
+    val name: String,
+    val description: String,
+    val budget: Int,
+    val rating: Double,
+    val releaseDate: String,
+    val imagePath: String
+) {
+    fun toDomain(genres: List<GenresEntity>) = MovieDetailsModel(
```

Comment:
It is best to avoid passing parameters to mappers as it makes relations between models and could be difficult to maintain in the future, maybe some other approach could be used here?

Reply:
I found a new approach with this :D